server/monitor.go

```diff
@@ -0,0 +1,504 @@
+	ServerID   string             `json:"server_id"`
+	Now        time.Time          `json:"now"`
+	Type       string             `json:"type"`
+	Limits     stores.StoreLimits `json:"limits"`
+	TotalMsgs  int                `json:"total_msgs"`
+	TotalBytes uint64             `json:"total_bytes"`
+}
+
+// Clientsz lists the client connections
+type Clientsz struct {
+	ClusterID string     `json:"cluster_id"`
+	ServerID  string     `json:"server_id"`
+	Now       time.Time  `json:"now"`
+	Offset    int        `json:"offset"`
+	Limit     int        `json:"limit"`
+	Count     int        `json:"count"`
+	Total     int        `json:"total"`
+	Clients   []*Clientz `json:"clients"`
+}
+
+// Clientz describes a NATS Streaming Client connection
+type Clientz struct {
+	ID            string                      `json:"id"`
+	HBInbox       string                      `json:"hb_inbox"`
+	Subscriptions map[string][]*Subscriptionz `json:"subscriptions,omitempty"`
+}
+
+// Channelsz lists the name of all NATS Streaming Channelsz
+type Channelsz struct {
+	ClusterID string    `json:"cluster_id"`
+	ServerID  string    `json:"server_id"`
+	Now       time.Time `json:"now"`
+	Offset    int       `json:"offset"`
+	Limit     int       `json:"limit"`
+	Count     int       `json:"count"`
+	Total     int       `json:"total"`
+	Channels  []string  `json:"channels"`
+}
+
+// ChannelsWithSubsz list detailed channels along with subscriptions on each channel
+type ChannelsWithSubsz struct {
+	ClusterID string      `json:"cluster_id"`
+	ServerID  string      `json:"server_id"`
+	Now       time.Time   `json:"now"`
+	Offset    int         `json:"offset"`
+	Limit     int         `json:"limit"`
+	Count     int         `json:"count"`
+	Total     int         `json:"total"`
+	Channels  []*Channelz `json:"channels"`
+}
+
+// Channelz describes a NATS Streaming Channel
+type Channelz struct {
+	Name          string           `json:"name"`
+	Msgs          int              `json:"msgs"`
+	Bytes         uint64           `json:"bytes"`
+	FirstSeq      uint64           `json:"first_seq"`
+	LastSeq       uint64           `json:"last_seq"`
+	Subscriptions []*Subscriptionz `json:"subscriptions,omitempty"`
+}
+
+// Subscriptionz describes a NATS Streaming Subscription
+type Subscriptionz struct {
+	Inbox        string `json:"inbox"`
+	AckInbox     string `json:"ack_inbox"`
+	DurableName  string `json:"durable_name,omitempty"`
+	QueueName    string `json:"queue_name,omitempty"`
+	IsDurable    bool   `json:"is_durable"`
+	MaxInflight  int    `json:"max_inflight"`
+	AckWait      int    `json:"ack_wait"`
+	LastSent     uint64 `json:"last_sent"`
+	PendingCount int    `json:"pending_count"`
+	IsStalled    bool   `json:"is_stalled"`
+}
+
+func (s *StanServer) startMonitoring(nOpts *gnatsd.Options) error {
+	var hh http.Handler
+	// If we are connecting to remote NATS Server, we start our own
+	// HTTP(s) server.
+	if s.opts.NATSServerURL != "" {
+		s.natsServer = gnatsd.New(nOpts)
```

Comment:
When not embedding the NATS server, we would have to duplicate the code starting the http server, but also the checks about the http ports (a PR that was recently merged in NATS Server), which now happen in gnatsd.StartMonitoring(). It is not *that* much code, but if we can avoid duplicating, I think there is merit. I am not so concern with the heavyweight of creating a new instance of NATS Server (the server is not started, not client accept loop, etc..), but I agree that the NATS Server specific endpoints would then always return empty values.